Create toCharArray method (#1624)
Codecademy/docs: content/java/concepts/strings/terms/toCharArray/toCharArray.md

---
Title: '.toCharArray()'
Description: 'Returns an new character array from a given string.'
Subjects:

>   - 'Java: Strings'

Reviewer: ```suggestion
  - 'Computer Science'
```

For reference, you can check out the [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) file for future entry's subject field metadata 🙂

Tags:
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The method `.toCharArray()` returns a new character array from the given string. The length of the array is equal to the length of the original string.

Reviewer: ```suggestion
The method **`.toCharArray()`** returns a new character array from the given string. The length of the array is equal to the length of the original string.
```

## Syntax

> ```java

Reviewer: ```suggestion
```pseudo
```

string.toCharArray()
```

> - Where `string` is the string to be transformed to a new array of characters.

Reviewer: ```suggestion

- Where `string` is the string to be transformed to a new array of characters.
```

- This method doesn't take any parameters.

> 
> 

Reviewer: ```suggestion
```

## Example

> The example below uses the `.toCharArray()` method to create a new array, and uses a loop to iterate through the array and print each character:

Reviewer: ```suggestion
The example below uses the `.toCharArray()` method to create a new array, uses a loop to iterate through the array , and prints each character:
```

```java
// Example.java
class Example {
  public static void main(String[] args) {

>     String salutation = "Hello CodeCademy!";

Reviewer: ```suggestion
    String salutation = "Hello Codecademy!";
```

    char[] textArray = salutation.toCharArray();

    for (int i = 0; i < textArray.length; i++) {
        System.out.println(textArray[i]);
      }
  }
}
```

This will produce the following output:

> ```

Reviewer: ```suggestion
```shell
```

H
e
l
l
o

C
o
d
e

> C

Reviewer: ```suggestion
c
```

---
Title: '.toCharArray()'
Description: 'Returns an new character array from a given string.'
Subjects:
  - 'Java: Strings'
Tags:
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-java'
---

The method `.toCharArray()` returns a new character array from the given string. Its length is equal to the the length of this string.

## Syntax

```java
string.toCharArray()
```
- Where `string` is the string to be transform in a new array of characters.
- This method doesn't take parameters.

## Example 1

Use `.toCharArray()` to print the character array `"Hello CodeCademy!"` and to demonstrate that the length is equal to the string's length:

```java
class Example {
  public static void main(String[] args) {
    String string = "Hello CodeCademy!";

    char[] array = string.toCharArray();

    System.out.println(array);
    System.out.println("The length of the string is: "+ string.length());
    System.out.println("The length of the array is: "+ array.length);
  }
}
```

The output would be:

```
Hello CodeCademy!
The length of the string is: 17
The length of the array is: 17

> ```

Reviewer 2: I would recommend deleting the first example. This is a pretty basic method and the additional example isn't really demonstrating some alternative functionality.

## Example 2

> Use `.toCharArray()` and then print all the characters from the array `"Hello CodeCademy!"` using a for loop:

Reviewer 2: ```suggestion
The example below uses the `.toCharArray()` method to create a new array, and uses a loop to iterate through the array and print each character:
```

> ```java

Reviewer 2: ```suggestion
```java
// Example2.java
```

class Example2 {
  public static void main(String[] args) {

>     String string = "Hello CodeCademy!";

Reviewer 2: ```suggestion
    String salutation = "Hello CodeCademy!";
```

>     char[] array = string.toCharArray();

Reviewer 2: ```suggestion
    char[] textArray = salutation.toCharArray();
```

>     for (int i = 0; i < array.length; i++) {
>         System.out.println(array[i]);

Reviewer 2: ```suggestion
    for (int i = 0; i < textArray.length; i++) {
        System.out.println(textArray[i]);
```

      }
  }
}
```

> The output would be:

Reviewer 2: ```suggestion
This will produce the following output:
```
